File: Easy/Valid_Parentheses/Valid_Parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> myStack = new Stack<>();
        Map<Character, Character> closeBracket = new HashMap<>();
        closeBracket.put(')', '(');
        closeBracket.put(']', '[');
        closeBracket.put('}', '{');

        for (char c : s.toCharArray()) {
            if (closeBracket.containsKey(c)) {
                if (!myStack.isEmpty() && myStack.peek() == closeBracket.get(c)) {
                    myStack.pop();
                } else {
                    return false;
                }
            } else {
                myStack.push(c);
            }
        }

        return myStack.isEmpty();

    }
}

File: Easy/Valid_Parentheses/Valid_Parentheses.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        let stack = [];
        let closeBracket = {
            ')': '(',
            ']': '[',
            '}': '{'
        }

        for (let c of s) {
            if (closeBracket[c]) {
                if (stack.length > 0 && stack[stack.length - 1] == closeBracket[c]) {
                    stack.pop();
                } else {
                    return false;
                }
            } else {
                stack.push(c);
            }
        }

        return stack.length == 0;
    }
}


